iceoryx_utils/include/iceoryx_utils/posix_wrapper/file_lock.hpp

```diff
@@ -45,7 +45,7 @@ enum class FileLockError
     INTERNAL_LOGIC_ERROR,
 };
 
-constexpr char PATH_PREFIX[] = "/var/lock/";
+constexpr char PATH_PREFIX[] = "/tmp/";
```

Comment:
According to the [file system hierarchy standard](https://en.wikipedia.org/wiki/Filesystem_Hierarchy_Standard) defined by the Linux foundation, lock files [shall](https://refspecs.linuxfoundation.org/FHS_3.0/fhs/ch05s09.html) be placed under `/var/locks`. Seems like your distribution decided to choose another location. Mac OS prefers [`/var/tmp/`](https://github.com/eclipse-iceoryx/iceoryx/pull/598/files#diff-65a31f3463e5bc0dce797aecc310342c1fbf2487fc602dcb8bd80019d5257640R48-R52), this will be available with #598 

QNX advises to use [`/var/locks`](http://www.qnx.com/developers/docs/qnxcar2/index.jsp?topic=%2Fcom.qnx.doc.neutrino.user_guide%2Ftopic%2Ffiles_var.html), but in our experience so far, it depends to the image, that is created.

Could you please add this with an `#ifdef` for common platforms? `installation.md#QNX` currently states that `/var/locks` is used, in case you change it please update it there.

Reply:
@mossmaurice sorry I read your comment after I merged the pull request :( ... but I will handle this comment nevertheless.

> According to the file system hierarchy standard defined by the Linux foundation, lock files shall be placed under /var/locks

This is correct but the access rights of `/var/lock` is in Gentoo, ArchLinux and OpenSuSe like the following:
```
drwxrwxr-x  5 root root    120 Mar 22 15:26 lock
# /var
```

This means the user which is starting RouDi has to be at least in the root group. On the other hand `/tmp/` is for all users always read and writable and since the applications and roudi has started as an arbitrary user it is the perfect spot for the lock file in QNX, Mac OS and Linux. If you insist in using `/var/lock` for the lock file we have to do the following things.

1. Write `#ifdef` for the platforms, instead of using `/tmp` which works for every platform (except Windows of course).
2. Add a chapter in the `installation.md` on howto add the user which is executing roudi to the lock management group of Linux, Mac OS (this will be a pain) and QNX.


If you do not object I would let the directory be `/tmp` and would add this to the `installation.md` since the file system hierarchy standard states the directory structure of a linux system and not where an application has to store the files. I admit it would be best to use the folders accordingly but it seems that on some linux systems we do not have access to `/var/lock` but we will always have access to `/tmp`.

By the way this also goes for QNX. The QNX boards of the past where also exactly in this way configured that everything was read only and we only had write access to `/tmp`.